packages/react-data-grid-react-window/src/components/DataGridBody/useDataGridBody.tsx

```diff
@@ -54,6 +59,28 @@ export const useDataGridBody_unstable = (
     [ariaRowIndexStart, children]
   );
 
+  const onScroll = React.useCallback(() => {
+    if (bodyRef.current && headerRef.current) {
+      headerRef.current.scroll({
+        left: bodyRef.current.scrollLeft,
+        behavior: 'instant',
+      });
+    }
+  }, []);
+
+  // Use a onScroll callback on the outerElement since react-window's onScroll will only work for horizontal scrolls
+  const setupOuterRef = React.useCallback(
+    (outerElement: HTMLElement | null) => {
+      bodyRef.current?.removeEventListener('scroll', onScroll);
+
+      bodyRef.current = outerElement;
+      if (outerElement) {
+        outerElement.addEventListener('scroll', onScroll);
```

Comment:
This event listener on the other hand needs to be removed on ref disposal. Easiest way to do this is to return a callback ref from `useMemo` so that you can keep a handle to the element on ref dispose - smth along the lines of this...

```ts
const ref = React.useMemo(() => {
  let dispose = () => null;

  return (el) => {
      if (el) {
        el.addEventListener();
        dispose = () => el.removeEventListener()
      } else {
        dispose();
      }
   }
})
```